### ISSUE TITLE
Request to add complete names of collections when the tooltip is opened in explore page 

### ISSUE THREAD

**Issue by pgundluru** [ImagingDataCommons/IDC-WebApp#1323]
This was requested by Andrey - Currently we have a new tool tip feature but some of the collections are just abbreviations and it would be helpful to have the full name shown when a user selects that collection. 
![image](https://github.com/ImagingDataCommons/IDC-WebApp/assets/85526819/ea981310-e702-448e-b45a-6b34b5c1a187)

**Comment by fedorov** [comment]
First, we need to define the full name of the program - those do not exist anywhere in tables. Then we should discuss if we want to display those in the list, or only in the tooltips.

**Comment by fedorov** [comment]
Need to think what we want to accomplish - ideally, full text search over short name, long name (to be determined) and (perhaps) descriptions as well? This would require rewriting of the search operation, which is implemented in JS.

**Comment by fedorov** [comment]
We could consider something similar to what GDC does with the search box. Clicking any of the items found would bring user to the dedicated project page (they also have dedicated pages for individual cases).

<img width="796" alt="Image" src="https://github.com/user-attachments/assets/65e57d61-0044-4a6d-957f-378c128b090a" />

**Comment by fedorov** [comment]
This is the document where I think we mapped program names to long names: https://docs.google.com/spreadsheets/d/1-sk8CMTDDj-deKv7sXglLvHUhDSNS1cRqUg5Oy5UpRY/edit?usp=sharing

**Comment by fedorov** [comment]
I see long program names here

<img width="805" alt="Image" src="https://github.com/user-attachments/assets/a498961a-64ef-4f7a-b063-0c43cbf2c3d3" />

**Comment by fedorov** [comment]
Let's do a hover.

**Comment by fedorov** [comment]
@bcli4d I think those long names should first be ingested into some BQ table?

**Comment by bcli4d** [comment]
Also, I see that in original_collections_metadata, the program for CMB-* collections is 'CMB', rather than 'Cancer Moonshot Biobank'. That will probably get fixed as a result of whatever we decide to do here.

**Comment by bcli4d** [comment]
This issue seemed to start off concerned about collection 'full names' but then seems to have evolved to program names.

So regarding collection names/IDs:
The analysis_results_metadata table has both an 'ID', e.g. _PROSTATEx-Seg-HiRes_ and a 'Title', e.g. _High Resolution Prostate Segmentations for the ProstateX-Challenge_, whereas original_collections_metadata and dicom_all has just an ID (in two forms, _collection_name_ and _collection_id_) but no Title/full name. So we have _CMMD_ and _cmmd_ but not _The Chinese Mammography Database_.

FWIW, this dichotomy arose because we used to scrape the TCIA Analysis results page, which used to include a Title, and the TCIA Collections page which never included a Title. Anyway, we can, of course, add a collection title if desired.

As a reminder, the _collection_name_ is the spelling/capitalization that the TCIA/NBIA APIs expect and that the webapp displays. The _collection_id_ is spelling that the webapp uses internally.

**Comment by s-paquette** [comment]
@bcli4d Yes, this has been morphed to include both pieces of information which would be shown in different places (but you don't need to worry about that part).

For the collections, if we could add the 'title' for the original collections metadata (so it's the same column in both tables) that would work best for me. 

For the programs, maintaining a separate table I would pull from (the way I do from original collections metadata) would be preferable. Program short name/ID and then full program name (or title) would be all I'd need there, with the understanding the short name/ID needs to be a form which like collection_id has no spaces in it (upper/lower case is fine).

**Comment by bcli4d** [comment]
Should the program table continue to be non-public? I don't see any real value to end users if both the program ID and name are in original_collections_metadata.
